server/src/main/java/org/elasticsearch/action/admin/indices/validate/query/ValidateQueryResponse.java

```diff
@@ -96,4 +105,34 @@ public void writeTo(StreamOutput out) throws IOException {
         }
 
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(VALID_FIELD, isValid());
+        RestActions.buildBroadcastShardsHeader(builder, params, this);
```

Comment:
shall we leverage addCustomFields here too and just move the valid field after the shards header?

Reply:
I feel like we are putting the `validate` field at the beginning on purpose https://www.elastic.co/guide/en/elasticsearch/reference/current/search-validate.html ? as it's for validation and we want to see the validate field at first ? If we modify the order here, will it impact elsewhere when it calls toXContent() of ValidateQueryResponse ?

Reply:
I see, but you can't rely on keys ordering when parsing json... we are free to change order as that is not guaranteed. Maybe we should then update the docs as well accordingly just to make sure that nobody thinks we guarantee that valid comes first.